File: mpi_for_python/slides/helloworld.c
#include <mpi.h>
#include <stdio.h>

int main(int argc, char *argv[]) {
  int rank, size, len;
  char name[MPI_MAX_PROCESSOR_NAME];

  MPI_Init(&argc, &argv);
  MPI_Comm_rank(MPI_COMM_WORLD, &rank);
  MPI_Comm_size(MPI_COMM_WORLD, &size);
  MPI_Get_processor_name(name, &len);
  printf("Hello, World! "
         "I am process %d of %d on %s\n",
         rank, size, name);
  MPI_Finalize();
  return 0;
}

File: mpi_for_python/slides/helloworld.py
from mpi4py import MPI

rank = MPI.COMM_WORLD.Get_rank()
size = MPI.COMM_WORLD.Get_size()
name = MPI.Get_processor_name()

print ("Hello, World! "
       "I am process {} of {} on {}"
       .format(rank, size, name))
